### ISSUE TITLE
why atk image's color is abnormal when use data normalization in transform

### ISSUE THREAD

**Issue by muyuuuu** [Harry24k/adversarial-attacks-pytorch#78]
my dataloader:

```py
elif name == "cifar10":
    transform = torchvision.transforms.Compose([
        torchvision.transforms.RandomCrop(32, padding=4),
        torchvision.transforms.RandomHorizontalFlip(),
        torchvision.transforms.ToTensor(),
        torchvision.transforms.Normalize((0.5, 0.48, 0.45), (0.25, 0.24, 0.26))
    ])

    train_dataset = torchvision.datasets.CIFAR10(root=data_path,
                                                 train=True,
                                                 download=True,
                                                 transform=transform)
    test_dataset = torchvision.datasets.CIFAR10(root=data_path,
                                                train=False,
                                                download=True,
                                                transform=transform)
```

![image](https://user-images.githubusercontent.com/43681138/171015250-e12ff373-38f1-4143-bf12-5acc43e30852.png)

Use `fgsm`, `pgd`, 'cw' to generate attack data, but the color may be abnormal. After read #11 ,

> To make it easy to use adversarial attacks, a reverse-normalization is not included in the attack process. To apply an input normalization, please add a normalization layer to the model.

why do this? `torchvision.transforms.Normalize((0.5, 0.48, 0.45), (0.25, 0.24, 0.26))` also can scale the value range of image to [0, 1]?

**Comment by muyuuuu** [comment]
![atk_fgsm_cifar10](https://user-images.githubusercontent.com/43681138/171029206-b04ff18e-d5f5-4aee-be62-3ca069748789.png)

I get normally fgsm attack image data as describtion in #11 , but why?

**Comment by Harry24k** [comment]
The epsilon in FGSM, PGD, and CW is normally set to 2/255 or 4/255 or 8/255, which are the values for an image in the range [0, 1].
Thus, all images should be scaled to [0, 1] with transform[to.Tensor()] before used in attacks.

**Comment by muyuuuu** [comment]
> Thus, all images should be scaled to [0, 1] with transform[to.Tensor()] before used in attacks.

Oh! I know what you said why to do this. thank you very much.
